app/experimenter/static/js/components/DesignForm.js

```diff
@@ -56,12 +56,31 @@ class DesignForm extends React.PureComponent {
     });
   }
 
-  handleDataChange(key, value) {
+  async handleDataChange(key, value) {
     this.setState(({ data }) => ({
       data: data.set(key, value),
+      loading: true,
     }));
   }
 
+  async handleMultiPrefChange(key, value) {
+    this.setState(
```

Comment:
ohh we're going forward with click button -> rest call already :)

Reply:
yes, @jaredkerim thought I should go with this approach here. Or else, I would have had to add the blank preferences on the client side, which is kind of out of pattern at this point. :)

Reply:
It didn't turn out too bad, me thinks.

Reply:
Oh and it occurred to me we should stick to the pattern of always calling methods that modify state on components with `handle` at the start, and also I think `reload` is probably a better word than `refresh` here, and we should probably mention that this calls the API, so I think `handleReloadAPIState` is probably what we want.  Thanks for humouring me @uhlissuh ! Sorry to be such a pain!  This change is really succinct, it looks great!

![giphy](https://user-images.githubusercontent.com/119884/70491302-ea9d8b80-1ace-11ea-990b-c38cf5cb2c95.gif)